File: report-viewer/src/model/Cluster.ts
/**
 * Represents a cluster identified by the clustering algorithm.
 * A cluster is a group of similar submissions.
 * @property averageSimilarity - The average similarity of the submissions in the cluster.
 * @property strength - The strength of the cluster
 * @property members - The ids of the submissions in the cluster
 */
export interface Cluster extends ReportFormatCluster {
  index: number
}

export interface ReportFormatCluster {
  averageSimilarity: number
  strength: number
  members: string[]
}

File: reporting/src/main/java/de/jplag/reporting/reportobject/model/Cluster.java
package de.jplag.reporting.reportobject.model;

import java.util.List;

/**
 * Represents a cluster with its average similarity, strength, and member identifiers.
 * @param averageSimilarity the average similarity score within the cluster
 * @param strength the strength metric of the cluster
 * @param members the list of members (submission IDs) in the cluster
 */
public record Cluster(double averageSimilarity, double strength, List<String> members) {
}
